### PR TITLE
Add serializer

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

This RP will add missing serializer part.

I am targeting this branch, because is PATCH.

## Changelog

```markdown
### Added
- Added `Sonata\Form\Serializer\BaseSerializerHandler`
- Added `Sonata\Form\Serializer\BaseSerializerHandlerInterface` 
```

### CODE REVIEW

File: docs/reference/installation.rst [hunk 1]
```diff
@@ -63,6 +63,19 @@ corresponding configuration node accordingly:
         sonata_form:
             form_type: horizontal
 
+If you are using jms/serializer subscribing methods you will need to configure the
```
Comment (greg0ire) [inline docs/reference/installation.rst:66]:
```suggestion
If you are using ``jms/serializer`` subscribing methods you will need to configure the
```

File: src/Bridge/Symfony/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -62,4 +63,25 @@ private function addFlashMessageSection(ArrayNodeDefinition $node): void
             ->end()
         ;
     }
+
+    /**
+     * Returns configuration for serializer formats.
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/Configuration.php:68]:
```suggestion
     * Adds configuration for serializer formats.
```

File: src/Bridge/Symfony/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -62,4 +63,25 @@ private function addFlashMessageSection(ArrayNodeDefinition $node): void
...
+     */
+    private function addSerializerFormats(ArrayNodeDefinition $node)
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/Configuration.php:70]:
```suggestion
    private function addSerializerFormats(ArrayNodeDefinition $node): void
```

File: src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php [hunk 4]
```diff
@@ -51,5 +53,17 @@ public function load(array $configs, ContainerBuilder $container): void
...
 
         $container->setParameter('sonata.form.form_type', $config['form_type']);
+
+        $this->configureSerializerFormats($config);
+    }
+
+    /**
+     * @param array $config
+     */
+    public function configureSerializerFormats($config)
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php:63]:
```suggestion
    public function configureSerializerFormats($config): void
```

File: src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php [hunk 5]
```diff
@@ -51,5 +53,17 @@ public function load(array $configs, ContainerBuilder $container): void
         $loader->load('validator.xml');
 
         $container->setParameter('sonata.form.form_type', $config['form_type']);
+
+        $this->configureSerializerFormats($config);
+    }
+
+    /**
+     * @param array $config
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php:61]:
```suggestion
     * @param mixed[] $config
```

File: src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php [hunk 6]
```diff
@@ -51,5 +53,17 @@ public function load(array $configs, ContainerBuilder $container): void
...
 
         $container->setParameter('sonata.form.form_type', $config['form_type']);
+
+        $this->configureSerializerFormats($config);
+    }
+
+    /**
+     * @param mixed[] $config
+     */
+    public function configureSerializerFormats($config): void
```
Comment (phansys) [inline src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php:63]: Why this method needs to be public?

File: src/Serializer/BaseSerializerHandler.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+{
+    /**
+     * @var ManagerInterface
+     */
+    protected $manager;
+
+    /**
+     * @var string[]
+     */
+    protected static $formats;
```
Comment (phansys) [inline src/Serializer/BaseSerializerHandler.php:34]:
```suggestion
    protected static $formats = [];
```

File: src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php (hunk 6 above)
Comment (wbloszyk) [reply src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php:63]: I move it form CoreBundle and didn't check.

File: src/Serializer/BaseSerializerHandler.php (hunk 7 above)
Comment (wbloszyk) [reply src/Serializer/BaseSerializerHandler.php:34]: This is set by bundle config. I think this is good suggestion, becouse someone can use it without bundle load.

File: src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php [hunk 8]
```diff
@@ -51,5 +53,17 @@ public function load(array $configs, ContainerBuilder $container): void
...
 
         $container->setParameter('sonata.form.form_type', $config['form_type']);
+
+        $this->configureSerializerFormats($config);
+    }
+
+    /**
+     * @param mixed[] $config
+     */
+    private function configureSerializerFormats($config): void
```
Comment (phansys) [inline src/Bridge/Symfony/DependencyInjection/SonataFormExtension.php:63]:
```suggestion
    private function configureSerializerFormats(array $config): void
```

**Comment by greg0ire** [comment]
Thanks @wbloszyk !

**Comment by greg0ire** [comment]
:thinking: this is not really a patch, it's brand new code :P

**Pull request merged**